mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +130,35 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    # we don't want to decode times, since we're typically using dates
+    # outside the allowed range
+    kwargs['decode_times'] = False
+    # files differ along the Time dimension
+    kwargs['concat_dim'] = 'Time'
```

Comment:
So the warning would occur if the user tries to set them?  I agree this would be a good warning (or error) to raise.